docs/tutorial_fixtures.rst

```diff
@@ -0,0 +1,275 @@
+Tutorial 4: Using Test Fixtures
+===============================
+
+.. versionadded:: 3.9.0
+
+A fixture in ReFrame is a test that manages a resource of another test.
+Fixtures can be chained to create essentially a graph of dependencies.
+Similarly to test dependencies, the test that uses the fixture will not execute until its fixture has executed.
+In this tutorial, we will rewrite the OSU benchmarks example presented in :doc:`tutorial_deps` using fixtures.
+We will cover only the basic concepts of fixtures that will allow you to start using them in your tests.
+For the full documentation of the test fixtures, you should refer to the :doc:`regression_test_api` documentation.
+
+The full example of the OSU benchmarks using test fixtures is shown below with the relevant parts highlighted:
+
+.. literalinclude:: ../tutorials/fixtures/osu_benchmarks.py
+   :lines: 6-
+   :emphasize-lines: 6-43,53,67-68,85-86,106-107
+
+Let's start from the leaf tests, i.e. the tests that execute the benchmarks (:class:`osu_latency_test`, :class:`osu_bandwidth_test` and :class:`osu_allreduce_test`).
+As in the dependencies example, all these tests derive from the :class:`OSUBenchmarkTestBase`, where we define a fixture that will take care of generating the binaries of the tests:
+
+.. literalinclude:: ../tutorials/fixtures/osu_benchmarks.py
+   :lines: 58
+
+A test defines a fixture using the :func:`~reframe.core.pipeline.RegressionMixin.fixture` builtin and assigns it a name by assigning the return value of the builtin to a test variable, here ``osu_binaries``.
+This name will be used later to access the resource managed by the fixture.
+
+As stated previously, a fixture is another full-fledged ReFrame test, here the :class:`build_osu_benchmarks` which will take care of building the OSU benchmarks.
+Each fixture is associated with a scope.
+This practically indicates at which level a fixture is shared with other tests.
+There are four fixture scopes, which are listed below in decreasing order of generality:
+
+- ``session``: A fixture with this scope will be executed once per a ReFrame run session and will be shared across the whole run.
```

Comment:
```suggestion
- ``session``: A fixture with this scope will be executed once per ReFrame run session and will be shared across the whole run.
```
I think the "a" is not correct syntax, but I don't have a strong opinion.

Reply:
You're correct.